events/importer/kulke.py

```diff
@@ -939,12 +936,46 @@ def _save_recurring_superevent(self, recurring_group):
                     EventAggregateMember.objects.get_or_create(
                         event=event, event_aggregate=aggregate
                     )
+                # Remove any extra event aggregate members
+                EventAggregateMember.objects.filter(event_aggregate=aggregate).exclude(
+                    event__in=events
+                ).delete()
         for event in events:
             event.super_event = aggregate.super_event
         Event.objects.bulk_update(events, ("super_event",))
 
         return True
 
+    def _handle_removed_events(
+        self, elis_event_ids: Sequence[int], begin_date: datetime
+    ) -> None:
+        # Find Kulke events that are not referenced in the latest data from Elis and delete them.
+        count, deleted = (
+            Event.objects.filter(
+                data_source=self.data_source,
+                start_time__gte=begin_date,
+                super_event_type__isnull=True,
+            )
+            .exclude(origin_id__in=elis_event_ids)
+            .delete()
+        )
+        if count:
+            logger.debug("Deleted %d events and associated objects: %s", count, deleted)
+
+        # Find super events that no longer contain at least two events and delete them
+        count, deleted = (
+            Event.objects.exclude(super_event_type__isnull=True)
+            .annotate(aggregate_member_count=Count("aggregate__members"))
+            .filter(aggregate_member_count__lt=2)
+            .delete()
+        )
+        if count:
+            logger.debug(
+                "Deleted %d empty super events and associated objects: %s",
+                count,
+                deleted,
+            )
+
```

Comment:
AFAIK we don't delete events, but soft delete them instead. E.g. `some_event.soft_delete()`. Not sure what the case is with importers though, but I can't find any references in the code base to anything that actually deletes events, so I suppose it's the safer option.

Don't think there's a queryset method for soft deletion though (e.g. `Event.objects.all().soft_delete()`), so you'd have to either add one or do it locally. For reference, here's the `Event` model's `soft_delete` method:
```
    def soft_delete(self, using=None):
        self.deleted = True
        self.save(update_fields=("deleted",), using=using, force_update=True)
```